src/extract.rs

```diff
@@ -55,18 +55,31 @@ pub fn extract_light_occluders(
             &ViewVisibility,
         )>,
     >,
+    camera_query: Query<Entity, With<Camera>>,
 ) {
+    let mut occluders_len = 0;
+
     for (render_entity, light_occluder, transform, view_visibility) in &light_occluders_query {
-        if !view_visibility.get() {
-            continue;
+        if view_visibility.get() {
+            occluders_len += 1;
+
+            commands
+                .entity(render_entity)
+                .insert(ExtractedLightOccluder2d {
+                    half_size: light_occluder.half_size,
+                    center: transform.translation().xy(),
+                });
+        } else {
+            commands
+                .entity(render_entity)
+                .remove::<ExtractedLightOccluder2d>();
```

Comment:
The extract schedule works with a new set of data each time it runs, so there's no need to remove the ExtractedLightOccluder.